### PR TITLE
Istio support init commit

### PR DESCRIPTION
**Pull request by sudswasavi**
This commit introduces the Istio layer 1 changes in AKO.
The commit introduces the following:

- Istio informers vendoring
- Event handlers for 3 istio objects
- Full sync support for 3 istio objects

### CODE REVIEW

File: ako-operator/controllers/akoconfig_controller.go [hunk 1]
```diff
@@ -51,13 +51,20 @@ type AKOConfigReconciler struct {
...
 
 var objListOnce sync.Once
 
-func getObjectList() map[types.NamespacedName]runtime.Object {
+// func getObjectList() map[types.NamespacedName]runtime.Object {
+// 	objListOnce.Do(func() {
+// 		objectList = make(map[types.NamespacedName]runtime.Object)
+// 	})
+// 	return objectList
+// }
```
Comment (chauhanshubham) [inline ako-operator/controllers/akoconfig_controller.go:63]: stale code. We can remove this.

File: ako-operator/controllers/akoconfig_controller.go [hunk 2]
```diff
@@ -90,8 +97,8 @@ func removeFinalizer(finalizers []string, key string) (result []string) {
 // +kubebuilder:rbac:groups="rbac.authorization.k8s.io",resources=clusterrolebindings,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="policy",resources=podsecuritypolicies,verbs=get;list;watch;create;update;patch;delete
 
-func (r *AKOConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	ctx := context.Background()
+func (r *AKOConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// ctx := context.Background()
```
Comment (chauhanshubham) [inline ako-operator/controllers/akoconfig_controller.go:101]: we can remove this too.

File: internal/k8s/crdcontroller.go [hunk 3]
```diff
@@ -285,6 +287,173 @@ func (c *AviController) SetupAKOCRDEventHandlers(numWorkers uint32) {
 	return
 }
 
+// SetupAKOCRDEventHandlers handles setting up of AKO CRD event handlers
+func (c *AviController) SetupIstioCRDEventHandlers(numWorkers uint32) {
+	utils.AviLog.Infof("Setting up AKO Istio CRD Event handlers")
```
Comment (chauhanshubham) [inline internal/k8s/crdcontroller.go:292]: Comment update is required here.

File: helm/ako/values.yaml [hunk 4]
```diff
@@ -26,6 +26,7 @@ AKOSettings:
     labelValue: ""
   servicesAPI: false # Flag that enables AKO in services API mode: https://kubernetes-sigs.github.io/service-apis/. Currently implemented only for L4. This flag uses the upstream GA APIs which are not backward compatible 
                      # with the advancedL4 APIs which uses a fork and a version of v1alpha1pre1 
+  istioEnabled: "true" # This flag indicates to AKO that it should listen on Istio resources.
```
Comment (chauhanshubham) [inline helm/ako/values.yaml:29]: This should be false by default?

**Review by chauhanshubham — COMMENTED** [review]
lgtm. minor nits.

File: ako-operator/controllers/akoconfig_controller.go (hunk 2 above)
Comment (sudswasavi) [reply ako-operator/controllers/akoconfig_controller.go:101]: Done.

File: helm/ako/values.yaml (hunk 4 above)
Comment (sudswasavi) [reply helm/ako/values.yaml:29]: Done.

File: ako-operator/controllers/akoconfig_controller.go (hunk 1 above)
Comment (sudswasavi) [reply ako-operator/controllers/akoconfig_controller.go:63]: Done.

File: internal/k8s/ako_init.go [hunk 5]
```diff
@@ -463,6 +463,38 @@ func (c *AviController) FullSyncK8s() error {
 		}
 	}
 
+	// Section for Istio
+	if lib.IsIstioEnabled() {
+		// If Istio is enabled, we need to listen to few of the Istio CRDs.
```
Comment (monotosh-avi) [inline internal/k8s/ako_init.go:468]: I think we can add an info log in this section to denote that istio objects are being processed.

**Review by monotosh-avi — APPROVED** [review]
Looks good to  me.

File: internal/k8s/ako_init.go (hunk 5 above)
Comment (sudswasavi) [reply internal/k8s/ako_init.go:468]: Will add it in the next PR.

**Review by chauhanshubham — APPROVED** [review]

**Pull request merged**